aodndata/acorn/handler.py

```diff
@@ -0,0 +1,133 @@
+import os
+import re
+from datetime import datetime
+from netCDF4 import Dataset
+
+from aodncore.pipeline import HandlerBase, PipelineFilePublishType
+from aodncore.pipeline.exceptions import InvalidFileNameError
+from aodncore.util.misc import get_pattern_subgroups_from_string
+
+FILE_TYPE_NEED_INDEX = ('radial', 'radial_quality_controlled', 'gridded_1h-avg-current-map_non-QC',
+                        'gridded_1h-avg-current-map_QC', 'gridded_1h-avg-wind-map_QC',
+                        'gridded_1h-avg-wave-site-map_QC')
+
+ACORN_FILE_PATTERN = re.compile(r"""
+                                (.*/|)IMOS_ACORN_
```

Comment:
What's the reason for the extra wildcard stuff on the front of the regex? Would be cleaner if this was just the filename and things using the regex were expected to use the basename, because the directory shouldn't ever really be considered when matching